fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/CollectionObjectTest.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2016 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */package edu.unc.lib.dl.fcrepo4;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.isNull;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import com.hp.hpl.jena.rdf.model.Model;
+
+import edu.unc.lib.dl.fedora.ObjectTypeMismatchException;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.PcdmModels;
+
+public class CollectionObjectTest extends AbstractFedoraTest {
+	
+	private PID pid;
+	private CollectionObject collection;
+	
+	private PID folderChildPid;
+	private PID workChildPid;
+
+	private FolderObject folderChildObj;
+	private WorkObject workChildObj;
+
+	@Before
+	public void init() {
+		initMocks(this);
+
+		pid = PIDs.get(UUID.randomUUID().toString());
+
+		collection = new CollectionObject(pid, repository, dataLoader);
+
+		folderChildPid = PIDs.get(UUID.randomUUID().toString());
+		when(repository.mintContentPid()).thenReturn(folderChildPid);
+		
+		workChildPid = PIDs.get(UUID.randomUUID().toString());
+		when(repository.mintContentPid()).thenReturn(workChildPid);
```

Comment:
I don't think you need both of these `when(repository.mintContentPid()).thenReturn(...);` the second one will overwrite the first one.  If you want to repository.mindContentPid return two different specific pids in the same test, then you can chain like so: `when(repository.mintContentPid()).thenReturn(folderChildPid).thenReturn(workChildPid);` where you have one thenReturn per call (and I believe the last value will repeat for every call after the second))